### PR TITLE
Ft update subject relation

### PR DESCRIPTION
**Pull request by zhu327**
## 变更点(Changes)

- 缩减subject_relation表字段

## 相关issues (Which issues this PR fixes)

- Fixes #https://github.com/TencentBlueKing/bk-iam-saas/issues/1075

### CODE REVIEW

File: pkg/abac/pap/group.go [hunk 1]
```diff
@@ -29,21 +28,180 @@ import (
...
+	count, err := c.service.GetMemberCount(parentPK)
+	if err != nil {
+		return 0, errorWrapf(err, "service.GetMemberCount parentPK=`%s`", parentPK)
+	}
+
+	return count, nil
+}
+
+func (c *groupController) ListPagingMember(_type, id string, limit, offset int64) ([]SubjectMember, error) {
+	errorWrapf := errorx.NewLayerFunctionErrorWrapf(GroupCTL, "GetMemberCount")
```
Comment (wklken) [inline pkg/abac/pap/group.go:138]: `GetMemberCount`?

File: pkg/service/group.go [hunk 2]
```diff
@@ -125,22 +122,20 @@ func (l *groupService) ListEffectThinSubjectGroups(
 
 // ListSubjectGroups ...
```
Comment (wklken) [inline pkg/service/group.go:123]: 这几个函数声明不超过 120 字符的话, 改成一行

File: pkg/service/types/subject.go [hunk 3]
```diff
@@ -22,17 +22,14 @@ type Subject struct {
...
-	Type            string    `json:"type"`
-	ID              string    `json:"id"`
+	SubjectPK       int64     `json:"subject_pk"`
 	PolicyExpiredAt int64     `json:"policy_expired_at"`
 	CreateAt        time.Time `json:"created_at"`
 }
 
 // SubjectGroup subject关联的组
 type SubjectGroup struct {
 	PK              int64     `json:"pk"`
```
Comment (wklken) [inline pkg/service/types/subject.go:32]: 这个 PK 是 group subjectPK?

File: pkg/service/types/subject.go [hunk 4]
```diff
@@ -22,17 +22,14 @@ type Subject struct {
 // SubjectMember ...
 type SubjectMember struct {
 	PK              int64     `json:"pk"`
-	Type            string    `json:"type"`
-	ID              string    `json:"id"`
+	SubjectPK       int64     `json:"subject_pk"`
```
[the remark this answers is not in the document]
Comment (wklken) [reply pkg/service/types/subject.go:25]: 改成GroupMember?

[the remark this answers is not in the document]
Comment (zhu327) [reply pkg/service/types/subject.go:25]: pk统一成relation表本身的pk, groupMember中存在subjectPK, subjectGroup中新增parentPK

File: pkg/abac/pap/types.go [hunk 5]
```diff
@@ -25,7 +27,18 @@ type SubjectDepartment struct {
...
-	Type            string `json:"type"`
-	ID              string `json:"id"`
-	PolicyExpiredAt int64  `json:"policy_expired_at"`
+	PK              int64     `json:"pk"`
+	Type            string    `json:"type"`
+	ID              string    `json:"id"`
+	PolicyExpiredAt int64     `json:"policy_expired_at"`
+	CreateAt        time.Time `json:"created_at"`
+}
+
```
[the remark this answers is not in the document]
Comment (zhu327) [reply pkg/abac/pap/types.go:36]: pk统一成relation表的PK

File: pkg/abac/pap/group.go [hunk 6]
```diff
@@ -205,23 +329,85 @@ func (c *groupController) DeleteSubjectMembers(
...
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				continue
+			}
+
+			return nil, err
+		}
+
+		groups = append(groups, SubjectGroup{
+			PK:              m.PK,
```
[the remark this answers is not in the document]
Comment (zhu327) [reply pkg/abac/pap/group.go:380]: 修改为relation表pk

File: pkg/service/types/subject.go (hunk 3 above)
Comment (zhu327) [reply pkg/service/types/subject.go:32]: relation pk

File: pkg/service/group.go (hunk 2 above)
Comment (zhu327) [reply pkg/service/group.go:124]: 125

File: pkg/abac/pap/group.go [hunk 7]
```diff
@@ -29,21 +28,180 @@ import (
...
+	if err != nil {
+		return nil, errorWrapf(
+			err, "service.ListExistSubjectsBeforeExpiredAt subjectPKs=`%+v`, expiredAt=`%d` fail",
+			subjectPKs, expiredAt,
+		)
+	}
+
+	existSubjects := make([]Subject, 0, len(existSubjectPKs))
+	for _, pk := range existSubjectPKs {
+		subject, err := cacheimpls.GetSubjectByPK(pk)
```
[the remark this answers is not in the document]
Comment (zhu327) [reply pkg/abac/pap/group.go:78]: 30min

**Pull request merged**